dispatches/models/nuclear_case/chgtank.py

```diff
@@ -0,0 +1,590 @@
+        def enthalpy_balances(b, t):
+            return (
+                sum(b.energy_accumulation[t, p] for p in phase_list) == (
+                    sum(b.control_volume.properties_in[t].\
+                        get_enthalpy_flow_terms(p) for p in phase_list)
+                    - sum(b.control_volume.properties_out[t].\
+                        get_enthalpy_flow_terms(p) for p in phase_list)
+                    + b.heat_duty[t]))
+
+        # energy holdup calculation
+        @self.Constraint(self.flowsheet().config.time,
+                         phase_list,
+                         doc="Energy holdup integration")
+        def energy_holdup_integration(b, t, p):
+            return b.energy_holdup[t, p] == (
+                  sum(b.material_holdup[t, p, j]
+                      for j in component_list)
+                  * b.control_volume.properties_out[t].\
+                      enth_mol_phase[p])
+
+        # computing specific heat capacity ratio (cp/cv)
+        # NOTE: using ideal gas property Cv = Cp - R
+        @self.Expression(self.flowsheet().config.time,
+                         doc="specific heat ratio, cp/cv")
+        def gama(b, t):
+            for p in phase_list:
+                if p == 'Vap':
+                    return (
+                        b.control_volume.properties_in[t].\
+                            cp_mol_phase[p]
+                        / (b.control_volume.properties_in[t].\
+                           cp_mol_phase[p]
+                           - gas_constant)
+                        )
+                else:
+                    return 1
+
+        # Add constraint to determine the tank final temperature (T out)
+        # This constraint is derived from the 1st law of thermodynamics and 
+        # Ideal gas law assuming an adiabatic process
+        # Reference: Yang (2009). Int. J. of Hydrogen Energy 34, 6712-6721.
+
+        # TO DO: add pressure out equation for an isothermal process
+
+        @self.Expression(self.flowsheet().config.time,
+                         doc="Dimensionles time")
+        def dim_less_time(b, t):
+            return (
+                self.control_volume.properties_in[t].flow_mol
+                * b.dt[t]
+                / sum(b.initial_material_holdup[t, p, j]
+                      for (p, j) in pc_set)
+                )
+
+        @self.Constraint(self.flowsheet().config.time,
+                         doc="Tank Temperature calculations")
+        def tank_temperature_calculation(b, t):
+            return b.control_volume.properties_out[t].temperature == (
+                  b.gama[t] *
+                  b.control_volume.properties_in[t].temperature
+                  + ((b.initial_state[t].temperature
+                      - b.gama[t] *
+                      b.control_volume.properties_in[t].temperature)
+                     / (1 + b.dim_less_time[t])))
+
+    def initialize(blk, state_args={}, outlvl=idaeslog.NOTSET,
```

Comment:
Check the standard arguments here, and what version of IDAES we are working against. Notably, the latest main branch of IDAES has changed the defaults for the `solver` and `optarg` arguments.